src/cargopacket.h

```diff
@@ -39,15 +39,27 @@ extern SaveLoadTable GetCargoPacketDesc();
  */
 struct CargoPacket : CargoPacketPool::PoolItem<&_cargopacket_pool> {
 private:
+	/* A mathematical vector from (0,0). */
+	struct Vector {
+		int16_t x;
+		int16_t y;
+	};
+
 	uint16_t count{0}; ///< The amount of cargo in this packet.
 	uint16_t periods_in_transit{0}; ///< Amount of cargo aging periods this packet has been in transit.
 
 	Money feeder_share{0}; ///< Value of feeder pickup to be paid for on delivery of cargo.
 
 	TileIndex source_xy{INVALID_TILE}; ///< The origin of the cargo.
+	Vector travelled{0, 0}; ///< A vector (from 0,0) of how far the cargo travelled in a vehicle. Is a vector when in station. Is a temporary variable when in vehicle.
```

Comment:
Not sure what you mean with "from 0,0".
I got confused about the orientation of the vector. It is "source - destination", I would have expected the opposite :)
So:
* If inside station: Vector from current location to source tile.
* If inside vehicle: Intermediate value.

Reply:
The "from 0,0" is from a pedantic discussion that a vector needs an origin. If unspecified it is ofc 0,0, so the pedanticness confused you :D